controllers/api/v1/keyword_test.go

```diff
@@ -0,0 +1,168 @@
+package apiv1
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	_ "google-scraper/initializers"
+	. "google-scraper/tests"
+
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gstruct"
+	"github.com/tidwall/gjson"
+)
+
+type KeywordListResponse struct {
+	Id              int64
+	Name            string
+	SearchCompleted bool
+	CreatedAt       string
+	Meta            gjson.Result
+	Links           gjson.Result
+}
+
+var _ = Describe("KeywordController", func() {
+	Describe("GET /api/v1/keywords", func() {
+		Context("Given valid credentials", func() {
+			It("returns 200 status code", func() {
+				email := "john@example.com"
+				password := "secret"
+				user := FabricateUser("John", email, password)
+				FabricateKeyword("Buy domain", false, &user)
+				client := FabricateOAuthClient(uuid.New().String(), uuid.New().String())
+				oauthToken := FabricateOAuthToken(client, user.Id)
+				header := http.Header{"Authorization": {fmt.Sprintf("Bearer %v", oauthToken.GetAccess())}}
+
+				response := MakeAuthenticatedRequest("GET", "/api/v1/keywords", header, nil, nil)
+
+				Expect(response.StatusCode).To(Equal(http.StatusOK))
+			})
+
+			It("returns correct JSON response", func() {
+				email := "john@example.com"
+				password := "secret"
+				user := FabricateUser("John", email, password)
+				keyword := FabricateKeyword("Buy domain", true, &user)
+				client := FabricateOAuthClient(uuid.New().String(), uuid.New().String())
+				oauthToken := FabricateOAuthToken(client, user.Id)
+				header := http.Header{"Authorization": {fmt.Sprintf("Bearer %v", oauthToken.GetAccess())}}
+
+				response := MakeAuthenticatedRequest("GET", "/api/v1/keywords", header, nil, nil)
+				responseBody, err := ioutil.ReadAll(response.Body)
+				if err != nil {
+					Fail("Reading response body failed: " + err.Error())
+				}
+
+				links := gjson.Get(string(responseBody), "links")
+				meta := gjson.Get(string(responseBody), "meta")
+				tokenResponse := KeywordListResponse{
```

Comment:
Why is it named `tokenResponse`, did you mean:

```suggestion
				keywordResponse := KeywordListResponse{
```